source-map.bs

```diff
@@ -256,14 +364,166 @@ with four fields represent mapped code where a corresponding name does not exist
+            1. If |relativeOriginalPosition| is null, then:
+                1. If |relativeSourceIndex| is not null, [=optionally report an error=].
+                1. Continue with the next iteration.
+            <!-- TODO: If there is a relativeSourceIndex but not a relativeOriginalLine,
+                 should |sourceIndex| still be increased? -->
+            1. Increase |sourceIndex| by |relativeSourceIndex|.
+            1. Increase |originalLine| by |relativeOriginalLine|.
+            1. Increase |originalColumn| by |relativeOriginalPosition|.
+            1. If any of |sourceIndex|, |originalLine|, or |originalColumn| are less than 0, or if
+                |sourceIndex| is greater than or equal to |sources|'s [=list/size=], [=optionally
+                report an error=].
+            1. Else,
+                1. Set |decodedMapping|'s [=decoded mapping/originalSource=] to
+                    |sources|[|sourceIndex|].
+                1. Set |decodedMapping|'s [=decoded mapping/originalLine=] to |originalLine|.
+                1. Set |decodedMapping|'s [=decoded mapping/originalColumn=] to |originalColumn|.
+            1. [=Decode a base64 VLQ=] from |segment| given |position| and let |relativeNameIndex|
+                be the result.
+            1. If |relativeNameIndex| is not null, then:
+                1. Increase |nameIndex| by |relativeNameIndex|.
+                1. If |nameIndex| is negative or greater than |names|'s [=list/size=], [=optionally
+                    report an error=].
+                1. Else, set |decodedMapping|'s [=decoded mapping/name=] to |names|[|nameIndex|].
+            1. If |position| does not point to the end of |segment|, [=optionally report an
+                error=].
+    1. Increase |generatedLine| by 1.
+1. Return |decodedMappings|.
+
+To <dfn>decode a base64 VLQ</dfn> from a [=string=] |segment| given a [=position variable=]
+position, run the following stepsis the number :
+1. Let |first| be a [=byte=] whose the [=byte/value=] is the number corresponding to |segment|'s
+    |position|th [=code unit=], according to the [[base64]] encoding.
+
+    NOTE: The two most significant bits of |first| are 0.
+1. Let |sign| be 1 if |first| & 0x01 is 0x00, and -1 otherwise.
+1. Let |value| be (|first| &gt;&gt; 1) & 0x0F, as a number.
+1. Let |nextShift| be 4.
+1. Let |currentByte| be |first|.
+1. While |currentByte| & 0x20 is 0x20:
+    1. Advance |position| by 1.
+    1. If |position| points to the end of |segment|, throw an error.
+    1. Set |currentByte| to the [=byte=] whose the [=byte/value=] is the number corresponding to
+        |segment|'s |position|th [=code unit=], according to the [[base64]] encoding.
+    1. Let |chunk| be |currentByte| & 0x1F, as a number.
+    1. Add |chunk| &lt;&lt; |nextShift| to |value|.
+    1. If |value| is greater than or equal to 2<sup>31</sup>, throw an error.
+    1. Increase |nextShift| by 5.
+1. Advance |position| by 1.
+1. Return |value| * |sign|.
```

Comment:
There's a bug here when decoding `B` (which corresponds to -2147483648):
- `first = 1`
- `currentByte = 1`
- `value = 0`
- `sign = -1`
- `value * sign === -0`, not `-2147483648`

Reply:
Chrome's implementation also decodes `B` to 0 (it returns -0, since it uses floats): https://github.com/ChromeDevTools/devtools-frontend/blob/9b580b72f1334bb91778a052b0bc4bdc1eb8c452/front_end/core/sdk/SourceMap.ts#L846

Reply:
Reading https://tc39.es/source-map/#base64-vlq:

> Base64 VLQ: [[VLQ]](https://tc39.es/source-map/#biblio-vlq) is a [[base64]](https://tc39.es/source-map/#biblio-base64) value, where the most significant bit (the 6th bit) is used as the continuation bit, and the "digits" are encoded into the string least significant first, and where the least significant bit of the first digit is used as the sign bit.

It seems to me that, when decoding B (000001):
- the continuation bit is 0 (so this VLQ is complete)
- the sign bit is 1
- the digits are 0000

0000 is 0, so the absolute value of this number is 0.

Reply:
I did fix Chrome years ago: https://chromium-review.googlesource.com/c/v8/v8/+/2018882

Reply:
I opened https://github.com/tc39/source-map/issues/123 to continue the normative discussion, for now I'll update this PR to let implementations return _either_ 0 or -2147483648. My goal with this PR is to clarify what is already happening and what the spec already says, and not to force changes in implementations.